src/chapters/more-on-functions/recursion-walkthrough.rst

```diff
@@ -0,0 +1,207 @@
+
+Joining Array Elements With a Loop
+-----------------------------------
+
+Examine the code samples below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Use a ``for`` loop to iterate through the array and add each entry into the ``newString`` variable.
+     - Use a ``while`` loop to add the first element in the array to ``newString``, then remove that element from the array.
+
+   * - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         for (i = 0; i < arr.length; i++){
+            newString = newString + arr[i];
+         }
+
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         ['L', 'C', '1', '0', '1']
+
+     - .. sourcecode:: js
+
+         let arr = ['L', 'C', '1', '0', '1'];
+         let newString = '';
+
+         while (arr.length > 0){
+            newString += arr[0];
+            arr.shift();
+         }
+         console.log(newString);
+         console.log(arr);
+
+       **Output**
+
+       .. sourcecode:: js
+
+         'LC101'
+         [ ]
+
+Inside each loop, the code simply adds two strings together - whatever is
+stored in ``newString`` plus one element from the array. In the ``for`` loop,
+the element is the next item in the sequence of entries.  In the ``while``
+loop, the element is always the first entry from whatever remains in the array.
+
+OK, the loops join the array elements together. Now let's see how to
+accomplish the same task without a ``for`` or ``while`` statement.
+
+Bring In Recursion Concepts
+----------------------------
+
+First, state the problem to solve: *Combine the elements from an array into a
+string*.
+
+Second, split the problem into small, identical steps: Looking at the loops
+above, the "identical step" is just adding two strings together - ``newString``
+and the next entry in the array.
+
+Third, build a function to accomplish the small steps: Let's call the function
+``combineEntries``, and we will set an array as the parameter.
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      //some code here
+   }
+
+We want ``combineEntries`` to repeat over and over again until the task is
+complete.
+
+How do we make this happen without using ``for`` or ``while``?
+
+.. index:: ! base case
+
+Identifying the Base Case
+--------------------------
+
+| ``for`` and ``while`` loops end when a particular condition evaluates to
+   ``false``. In the examples above, these conditions are
+| ``i < arr.length`` and ``arr.length > 0``, respectively.
+
+With recursion, we do not know how many times ``combineEntries`` must be
+called. To make sure the code stops at the proper time, we need to identify a
+condition that ends the process. This is called the **base case**, and it
+represents the simplest possible task for our function.
+
+``if`` the base case is ``true``, the recursion ends and the task is complete.
```

Comment:
Code formatting is not needed for "if" on this line and the next. The word is being used colloquially here, and not in explicit reference to the `if` statement.

Reply:
I know, but I thought I'd apply a little artistic license here, especially since we need an ``if`` statement in the code.  You know, foreshadowing, a quality literary technique.  ;-)

``if`` you feel strongly about this, I'd be OK changing it back.